app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.kt

```diff
@@ -0,0 +1,167 @@
+package fr.free.nrw.commons.explore.recentsearches
+
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import fr.free.nrw.commons.R
+import fr.free.nrw.commons.databinding.FragmentSearchHistoryBinding
+import fr.free.nrw.commons.di.CommonsDaggerSupportFragment
+import fr.free.nrw.commons.explore.SearchActivity
+import javax.inject.Inject
+
+/**
+ * Displays the recent searches screen.
+ */
+class RecentSearchesFragment : CommonsDaggerSupportFragment() {
+
+    @set:Inject
+    var recentSearchesDao: RecentSearchesDao? = null
+    private var _binding: FragmentSearchHistoryBinding? = null
+    private val binding get() = _binding!!
+    private var recentSearches: List<String> = emptyList()
+    var adapter: ArrayAdapter<String>? = null
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentSearchHistoryBinding.inflate(inflater, container, false)
+
+        val recentSearches = recentSearchesDao?.recentSearches(10) ?: emptyList()
+
+        if (recentSearches.isEmpty()) {
+            binding.recentSearchesDeleteButton.visibility = View.GONE
+            binding.recentSearchesTextView.setText(R.string.no_recent_searches)
+        }
+
+        binding.recentSearchesDeleteButton.setOnClickListener {
+            showDeleteRecentAlertDialog(requireContext())
+        }
+
+        adapter = ArrayAdapter(
+            requireContext(), R.layout.item_recent_searches,
+            recentSearches
+        )
+        binding.recentSearchesList.adapter = adapter
+        binding.recentSearchesList.setOnItemClickListener { _, _, position, _ ->
```

Comment:
Discussion:
Just a followup feature suggestion while using the app. What if we add swipe right, swipe left action on the item to delete particular item? 

https://github.com/user-attachments/assets/e9138df6-0340-4330-989f-7a24a080a6af
~ Similar to like this deleting the item by performing swipe action.

Reply:
That would be a nice extra indeed.
Great to implement if it is short and maintainable. :-)